src/iris/api.py

```diff
@@ -1375,13 +1375,57 @@ def on_get(self, req, resp, incident_id):
         resp.body = payload
 
     def on_post(self, req, resp, incident_id):
+        '''
+        Claim incidents by incident id. Deactivates the incident and
+        any associated messages, preventing further escalation.
+
+        **Example request**:
+
+        .. sourcecode:: http
+
+           POST /v0/incidents/123 HTTP/1.1
+           Content-Type: application/json
+
+           {
+               "owner": "jdoe"
+           }
+
+        **Example response**:
+
+        .. sourcecode:: http
+
+           HTTP/1.1 200 OK
+           Content-Type: application/json
+
+           {
+               "incident_id": "123",
+               "owner": "jdoe",
+               "active": false
+           }
+
+        '''
         incident_params = ujson.loads(req.context['body'])
 
         try:
             owner = incident_params['owner']
         except KeyError:
             raise HTTPBadRequest('"owner" field required')
 
+        if owner is not None:
+            connection = db.engine.raw_connection()
+            cursor = connection.cursor()
+            cursor.execute(
+                '''SELECT EXISTS(
+                     SELECT 1
+                     FROM `target`
+                     WHERE `target`.`name` = %s
+                     AND `type_id` = (SELECT `id` FROM `target_type` WHERE `name` = 'user'))''',
+                owner)
+            owner_valid = cursor.fetchone()[0]
```

Comment:
can you add a `cursor.close()` here? I'll merge after that.

Reply:
Bump :)

Reply:
Bump :)

Reply:
Oops